File: microservices/auth-service/src/test/java/com/karate/authservice/it/config/PostgresTc.java
package com.karate.authservice.it.config;

import org.testcontainers.containers.PostgreSQLContainer;

public final class PostgresTc {
    private static final PostgreSQLContainer<?> INSTANCE =
            new PostgreSQLContainer<>("postgres:16-alpine")
                    .withDatabaseName("auth_db_test")
                    .withUsername("test")
                    .withPassword("test")
                    .withReuse(true);

    static {
        INSTANCE.start();
    }

    private PostgresTc() {
    }

    public static PostgreSQLContainer<?> getInstance() {
        return INSTANCE;
    }
}

File: microservices/feedback-service/src/test/kotlin/com/karate/feedback_service/it/config/PostgresTc.kt
package com.karate.feedback_service.it.config

import org.testcontainers.containers.PostgreSQLContainer

object PostgresTc {
    val instance: PostgreSQLContainer<*> =
        PostgreSQLContainer("postgres:16-alpine")
            .withDatabaseName("feedback_db_test")
            .withUsername("test")
            .withPassword("test")
            .withReuse(true)
            .also { it.start() }
}